documentation/source/codes/codes.rst

```diff
@@ -17,6 +17,27 @@ Any additional pluggable dependencies should have APIs created in a similar way.
 External Code Interfaces
 ------------------------
 
+MappedParameterFrames
+^^^^^^^^^^^^^^^^^^^^^
+
+`MappedParameterFrames` extend the base `ParameterFrame` to allow mapping to external codes variables through `ParameterMapping`_.
+Default values for external codes are also enabled for instances where there are many unmapped variables that are only exposed to experienced users through the :py:attr:`problem_settings` of a `Solver` which is a direct link to the external code's variables.
+
+ParameterMapping
+""""""""""""""""
+
+ParameterMapping is used to create a connection between ``bluemira`` parameters and parameters on any external program. At its most basic level it is a key-value mapping between two variable names. On top of the mapping, how the parameter value flows between ``bluemira`` and the external program is modified by the :py:attr:`send` and :py:attr:`recv` attributes.
+
+:py:attr:`send`
+    true - set bluemira parameter value as input to external code
+
+    false - use default value as input to external code
+
+:py:attr:`recv`
+    true - set external code result to the new value of the bluemira parameter
+
+    false - keep the original bluemira parameter value ignoring the external value
+
 FPIs
```

Comment:
We aren't really calling these 'File Program Interfaces' any more, are we? Maybe this is out of the scope of this issue, but there are a few things in this document which I think are out of date. E.g., the 'Pattern for external codes' header. I don't think we have constants or mapping files any more.

Should I open an issue to review/update this doc?

Reply:
If you would that would be great

Reply:
https://github.com/Fusion-Power-Plant-Framework/bluemira/issues/1606